shenyu-integrated-test/shenyu-integrated-test-http/src/test/java/org/apache/shenyu/integrated/test/http/combination/HystrixPluginTest.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.shenyu.common.dto.convert.rule.HystrixHandle;
+import org.apache.shenyu.common.enums.HystrixIsolationModeEnum;
+import org.apache.shenyu.common.enums.OperatorEnum;
+import org.apache.shenyu.common.enums.ParamTypeEnum;
+import org.apache.shenyu.common.enums.PluginEnum;
+import org.apache.shenyu.common.utils.JsonUtils;
+import org.apache.shenyu.integratedtest.common.AbstractPluginDataInit;
+import org.apache.shenyu.integratedtest.common.helper.HttpHelper;
+import org.apache.shenyu.integratedtest.common.result.ResultBean;
+import org.apache.shenyu.web.controller.LocalPluginController.RuleLocalData;
+import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+import java.util.HashSet;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.Stream;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public final class HystrixPluginTest extends AbstractPluginDataInit {
+
+    private static final String TEST_HYSTRIX_PATH = "/http/test/hystrix/pass";
+
+    private static final String TEST_HYSTRIX_BAD_REQUEST_PATH = "/http/test/hystrix/fallback";
+
+    @BeforeEach
+    public void setup() throws IOException {
+        String pluginResult = initPlugin(PluginEnum.HYSTRIX.getName(), "");
+        assertThat(pluginResult, is("success"));
+    }
+
+    @Test
+    public void testPass() throws IOException, ExecutionException, InterruptedException {
+        String selectorAndRulesResult =
+                initSelectorAndRules(PluginEnum.HYSTRIX.getName(), "", buildSelectorConditionList(), buildRuleLocalDataList(TEST_HYSTRIX_PATH));
+        assertThat(selectorAndRulesResult, is("success"));
+        Future<ResultBean> resp = this.getService().submit(() -> HttpHelper.INSTANCE.getFromGateway(TEST_HYSTRIX_PATH, ResultBean.class));
+        assertEquals(200, resp.get().getCode());
+    }
+
+    @Test
+    public void testFallbackBySmeaphore() throws IOException, ExecutionException, InterruptedException {
+        String selectorAndRulesResult =
+                initSelectorAndRules(PluginEnum.HYSTRIX.getName(), "", buildSelectorConditionList(), buildRuleLocalDataList(TEST_HYSTRIX_BAD_REQUEST_PATH));
+        assertThat(selectorAndRulesResult, CoreMatchers.is("success"));
+        Set<String> resultSet = new HashSet<>();
+        Type returnType = new TypeToken<Map<String, Object>>() {
+        }.getType();
+        Future<Map<String, Object>> resp = this.getService().submit(() -> HttpHelper.INSTANCE.getFromGateway(TEST_HYSTRIX_BAD_REQUEST_PATH, returnType));
```

Comment:
It would be better to change the variable name to `resp0`~

Reply:
ok